pallets/ah-migrator/src/lib.rs

```diff
@@ -199,6 +202,47 @@ pub struct BalancesBefore<Balance: Default> {
 
 pub type BalanceOf<T> = <T as pallet_balances::Config>::Balance;
 
+// TODO: replace by pallet_message_queue::ForceSetHead once the 2503 merged from master.
+/// Allows to force the processing head to a specific queue.
+pub trait ForceSetHead<O> {
+	/// Set the `ServiceHead` to `origin`.
+	///
+	/// This function:
+	/// - `Err`: Queue did not exist, not enough weight or other error.
+	/// - `Ok(true)`: The service head was updated.
+	/// - `Ok(false)`: The service head was not updated since the queue is empty.
+	fn force_set_head(weight: &mut WeightMeter, origin: &O) -> Result<bool, ()>;
+}
+
+impl ForceSetHead<AggregateMessageOrigin> for () {
+	fn force_set_head(
+		_weight: &mut WeightMeter,
+		_origin: &AggregateMessageOrigin,
+	) -> Result<bool, ()> {
+		Ok(true)
+	}
+}
+
+/// The priority of the DMP queue during migration.
+///
+/// Controls how the DMP (Downward Message Passing) queue is processed relative to other queues
+/// during the migration process. This helps ensure timely processing of migration messages.
+/// The default priority pattern is defined in the pallet configuration, but can be overridden
+/// by a storage value of this type.
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "stable2503", derive(DecodeWithMemTracking))]
```

Comment:
We could merge master 2503 in now. Only downside is that we can probably not decode storage snapshots from RPC anymore.